src/abio/node/io.cljs

```diff
@@ -1,59 +1,50 @@
 (ns abio.node.io
   (:require
-    [abio.io :as io]
-    [clojure.string :as string]))
+   [abio.io :as io]
+   [clojure.string :as string]))
 
-(defrecord BufferedReader [raw-read raw-close buffer pos]
+(defrecord BufferedReader [path opts fs] ; `opts` should contain :encoding and optionally :flag
   abio.io/IReader
   (-read [_]
-    (if-some [buffered @buffer]
-      (do
-        (reset! buffer nil)
-        (subs buffered @pos))
-      (raw-read)))
-  abio.io/IBufferedReader
-  (-read-line [this]
-    (if-some [buffered @buffer]
-      (if-some [n (string/index-of buffered "\n" @pos)]
-        (let [rv (subs buffered @pos n)]
-          (reset! pos (inc n))
-          rv)
-        (if-some [new-chars (raw-read)]
-          (do
-            (reset! buffer (str (subs buffered @pos) new-chars))
-            (reset! pos 0)
-            (recur this))
-          (do
-            (reset! buffer nil)
-            (let [rv (subs buffered @pos)]
-              (if (= rv "")
-                nil
-                rv)))))
-      (if-some [new-chars (raw-read)]
-        (do
-          (reset! buffer new-chars)
-          (reset! pos 0)
-          (recur this))
-        nil)))
+    (.readFileSync fs path (clj->js opts)))
+  (-read [_ cb]
+    ;; TODO if this is a fd it won't get closed automatically; eventually that needs to be checked for.
+    (.readFile fs path (clj->js opts) cb))
+
+  abio.io/IClosable
+  ;; Even though there's nothing to close here, we keep it to preserve the use of `with-open`
+  (-close [_] nil))
+
+(defrecord BufferedWriter [path opts fs]
+  abio.io/IAbioWriter
+  (-write [_ output]
+    (.writeFileSync fs path output (clj->js opts)))
+  (-write [_ output cb]
+    (.writeFile fs path output (clj->js opts) cb))
+
   abio.io/IClosable
   (-close [_]
-    (raw-close)))
+    nil))
 
-(defrecord Bindings [fs]
+(defrecord Bindings [fs sep]
   abio.io/IBindings
+  (-path-sep [this] sep)
   (-directory? [this f]
     (.. fs (lstatSync f) (isDirectory)))
-  (-list-files [this d])
+  (-list-files [this d]
+    (.. fs (readdirSync d)))
+  (-list-files [this d cb]
+    (.. fs (readdir d cb)))
   (-delete-file [this f])
-  (-file-reader-open [this path encoding]
-    (let [fd (.openSync fs path "r")
-          read-stream (.createReadStream fs nil #js {:fd fd :encoding encoding})]
-      (.pause read-stream)
-      (.read read-stream)                                   ; Hack to get buffer primed
-      (->BufferedReader #(.read read-stream) #(.close fs fd) (atom nil) (atom 0))))
-  (-file-reader-read [this reader])
-  (-file-reader-close [this reader]))
+
+  (-file-reader-open [this path opts]
+    (->BufferedReader path opts fs))
+
+  ;; Default to non-destructive write
+  (-file-writer-open [this path opts]
+    (->BufferedWriter path (merge {:flag "a"} opts) fs)))
 
 (defn bindings
   []
-  (->Bindings (js/require "fs")))
+  (->Bindings (js/require "fs")
```

Comment:
Oh I see the require here, I suspect you are passing it explicitly for testing but I haven't reached the end yet :)

Reply:
I just learned, actually, that you can do the following in cljs instead!

```clojure
(require 'fs '[path :refer [sep]])
```
Or, the `ns` declaration form
```clojure
(ns abio.node.walkthrough
  (:require [fs]
            [path :refer [sep]))
```

So, given that, I'm starting to rethink how to bring these node modules in.